### ISSUE TITLE
Cows get stuck, run into each other, don't eat

### ISSUE THREAD

**Issue by egorananyev** [capnkirok/animania#610]
### Expected Behavior
Cows not getting stuck, eating, drinking, and mating without assistance.

### Actual Behavior
Cows get stuck. They just pointlessly run into each other for no apparent reason. Sometimes they all run into a fence. The spots they pick to do so appear random. Even two animals within plenty of space is enough for this behaviour to manifest. Whether or not there is a trough doesn't make a difference. As a result, I am having to feed them manually, as they do not get their own food or water due to being stuck.

### Steps to Reproduce
Have at least two cows in an enclosure together.

### Version of Minecraft, version of Animania, Single Player or Server
animania-1.10.2-1.4.7
single-player

### Screenshots encouraged
![2018-04-21_13 14 50](https://user-images.githubusercontent.com/6405803/39080688-8dd6c3d8-4566-11e8-960a-0ea7356bc63c.png)
![2018-04-21_13 14 58](https://user-images.githubusercontent.com/6405803/39080689-8e0c1b5a-4566-11e8-8b31-ffc1656a1fc7.png)

**Comment by egorananyev** [comment]
Another example: Loghorn hens react toward food, but on all other times, they cluster in the corner. Doesn't seem to happen with Orpingtons.

Just to clarify: I've tried both the most recent (1.4.7) and the previous version of the mod (1.4.6). This issue is identical.

![2018-04-21_23 48 06](https://user-images.githubusercontent.com/6405803/39086068-edd2bb04-45be-11e8-95c9-2c3d35242d77.png)
![2018-04-21_23 48 00](https://user-images.githubusercontent.com/6405803/39086069-ee03c168-45be-11e8-9b68-027eb772b1b3.png)

**Comment by capnkirok** [comment]
This can happen when they are pathing to a closer goal. (By the way, all chickens use the same AI, by the way, so there is no difference between Leghorns and Orpingtons).

Cows for instance, like flowers, crops, and tall grass. They might have found a block that is closer than the trough and can't get to it. Hence they appear stuck. Try levelling the area near the cows (e.g., removing all grass and flowers) so that the only option is the trough. Also can try moving the trough. If this fixes it, let me know. I may need to tweak the AI again. Same rule can be applied to chickens. 

I haven't seen this posted in a while, I thought this was fixed, so please let me know what you find out.

**Comment by egorananyev** [comment]
I moved the cows to a different enclosure, and that fixed it for them. I also cleared the area around them of tall grass, but that wasn't necessary.

But with chickens, the behaviour persisted. I noticed that I need to exit and reload the world, and the issue gets momentarily solved. But then they return to the original behavior of being extremely specific about the spot they want to be in, for no apparent reason.

![2018-04-22_09 04 21](https://user-images.githubusercontent.com/6405803/39090815-380d7bb2-461b-11e8-9ae9-421fbee10695.png)

**Comment by capnkirok** [comment]
Thanks for the quick reply. Definitely helps, along with the screenshots. I can work with this :)

I have a feeling that the problem you have with the chickens is the location of the nest. Can you move it away from the dirt wall into a more open space? Can you confim?

**Comment by egorananyev** [comment]
No problem! Thank you for looking into this. I've investigated this a bit further. 

1. I confirmed that this indeed happens to both my chicken breeds: Leghorns and Orpingtons.

2. I moved the nest, as you asked, but this did not change the behaviour of the chickens.

![2018-04-23_20 22 37](https://user-images.githubusercontent.com/6405803/39126350-ac628e98-4734-11e8-9a6e-587b5978733e.png)

3. Indeed, the chickens appear to be looking for a nest, as seen in the screenshot. However, this affects _both genders_.

![2018-04-23_20 21 36](https://user-images.githubusercontent.com/6405803/39126531-4732a192-4735-11e8-906f-8828cdf46e23.png)

3. Restarting the world doesn't really change anything in their behaviour, as I thought previously.

4. I don't know if this is related (I saw a similar _resolved_ issue), but the chickens appear to be very confused about the whole concept of drinking water. These two chickens in the screenshot are "thirsty", and none of the other ones are. They are also the only ones who get stuck in the trough. No chickens appear to be able to drink by themselves -- they just hover over the trough or splash in the hole with water, all the while dying of thirst.

![2018-04-23_20 23 28](https://user-images.githubusercontent.com/6405803/39126645-8022980e-4735-11e8-94c1-a86d3a23ddf9.png)

5. When I moved the cows to a different enclosure, the issue disappeared, but it didn't work for chickens. I tried to keep everything the same, but this behaviour now seems to be specific to chickens. Although I do got an occassional glitch with the cows -- for example, the cows would only drink from the trough, but when the water runs out, they wouldn't drink from a water hole.

6. My sheep are completely self sufficient and appear to drink from the water hole no problem. Hallelujah!
